optimum/exporters/onnx/base.py

```diff
@@ -596,7 +596,8 @@ def generate_dummy_inputs(self, framework: str = "pt", **kwargs):
             and self.use_cache_branch is not False
             and "attention_mask" in dummy_inputs
         ):
-            past_length = dummy_inputs["past_key_values"][0][0].shape[2]
+            past_length = dummy_inputs["past_key_values"][0][0].shape[-2]
```

Comment:
Why?

Reply:
With multi-query attention, there is one dimension less (num_heads), so I used reversed indexing.

Reply:
Alright, as long as it works for everyone.

Reply:
Yes it is, seq_len is in the same position for every model if we count from a reversed order.